### PR TITLE
Ports "Surgery now goes slower before failing" from /tg/ and operating table buff

### PR DESCRIPTION
**Pull request by BrumalLake**
## About The Pull Request
Ports [tgstation/#46640](https://github.com/tgstation/tgstation/pull/46640)
Surgery now slows down up to 50% before having a chance to fail
Sterilizing reagents such as honey also speed up surgery
Surgery tables have been buffed up to a 1x multiplier
Hand mirrors now also apply for the self surgery bonus
Also fixes being able to overwrite multipliers if you're on both a table or stasis unit etc

## Why It's Good For The Game
No more screwing up tend wounds 3 times in a row with an 80% chance on a stasis unit while the patient is awake
Field surgery is much more viable than before, while sterilizers are also much more useful
Surgery tables are now better at doing surgery on than stasis units instead of just allowing some researched surgeries to be performed on them

## Changelog
:cl:
add: sterilizers speed up surgery above cap
add: hand mirrors help with self surgery
tweak: surgery scales speed before starting to fail
balance: operating tables give better surgery chances than stasis units
fix: getting the lowest surgery multiplier from a structure when being on multiple structures
/:cl:

### CODE REVIEW

**Comment by eeSPee** [comment]
So surgery no longer randomly fails?

**Comment by BrumalLake** [comment]
> 
> 
> So surgery no longer randomly fails?

When you have a decent chance to succeed, yes

File: code/modules/surgery/surgery.dm [hunk 1]
```diff
@@ -115,30 +115,30 @@
 	SSblackbox.record_feedback("tally", "surgeries_completed", 1, type)
 	qdel(src)
 
-/datum/surgery/proc/get_propability_multiplier(mob/user)
+/proc/get_speed_modifier(mob/M, mob/N)
```
Comment (Archanial) [inline code/modules/surgery/surgery.dm]: Is there any reason it is a global proc?

File: code/modules/surgery/surgery.dm [hunk 2]
```diff
@@ -115,30 +115,30 @@
...
 	var/propability = 0.3
-	var/turf/T = get_turf(target)
+	var/turf/T = get_turf(N)
 	var/selfpenalty = 0
 	var/sleepbonus = 0
-	if(target == user)
-		if(HAS_TRAIT(user, TRAIT_SELF_AWARE) || locate(/obj/structure/mirror) in view(1, user))
+	if(N == M)
```
Comment (Archanial) [inline code/modules/surgery/surgery.dm]: Those names are really unclear

File: code/modules/surgery/surgery.dm (hunk 1 above)
Comment (BrumalLake) [reply code/modules/surgery/surgery.dm]: well i couldnt get surgery_steps to access it before so i just cut it, tg did the same

File: code/modules/surgery/surgery.dm (hunk 2 above)
Comment (BrumalLake) [reply code/modules/surgery/surgery.dm]: T is already claimed by the turf, so i just took the closest letter to M

Comment (Archanial) [reply code/modules/surgery/surgery.dm]: Why not use target and user?

**Comment by BrumalLake** [comment]
code works from my testing, but if i have to look for cleaner ways to do this, i will

File: code/modules/surgery/surgery.dm (hunk 2 above)
Comment (BrumalLake) [reply code/modules/surgery/surgery.dm]: i honestly have no clue how it worked with target/user when it only had one parameter for user, and many other procs just use letters

**Comment by BrumalLake** [comment]
pr has reached its final form not gonna do anything more

File: code/modules/surgery/surgery_step.dm [hunk 3]
```diff
@@ -55,10 +55,35 @@
...
+	var/propability = 0.3
+	var/turf/T = get_turf(target)
+	var/selfpenalty = 0
+	var/sleepbonus = 0
+	if(target == user)
+		if(HAS_TRAIT(user, TRAIT_SELF_AWARE) || locate(/obj/structure/mirror) in view(1, user) || user.get_inactive_held_item() == /obj/item/handmirror)
+			selfpenalty = 0.4
+		else
+			selfpenalty = 0.6
+	if(target != CONSCIOUS)
```
Comment (Archanial) [inline code/modules/surgery/surgery_step.dm]:
```suggestion
	if(!target.is_conscious())
```

File: code/modules/surgery/surgery.dm (hunk 2 above)
Comment (bloons3) [reply code/modules/surgery/surgery.dm]: Longer and more descriptive names don't cost anything extra

**Comment by TerryTehCoder** [comment]
> add: hand mirrors help with self surgery

Is self surgery possible on normal carbon mobs like Humans? I thought self surgery was ruled out a long time ago. Alternatively, are you talking about exclusively IPC's, because as far as I know they are the only ones who can pull off self surgery. 

Or i'm a idiot and haven't been keeping up.

**Comment by BrumalLake** [comment]
> 
> 
> > add: hand mirrors help with self surgery
> 
> Is self surgery possible on normal carbon mobs like Humans? I thought self surgery was ruled out a long time ago. Alternatively, are you talking about exclusively IPC's, because as far as I know they are the only ones who can pull off self surgery.
> 
> Or i'm a idiot and haven't been keeping up.

some surgeries can be performed on self, such as soft organ manipulation, which doesn't require sawing your bones which is every part excluding your chest and head
amputation, dental implant, cavity implant, prosthetic replacement, viral bonding, vein threading, and embed removal are all self operable

**Comment by lordScrubling** [comment]
> 
> 
> If the surgery doesn't have 100% success, it shouldn't have 100% success.

they still have lower multipliers than other surgeries

**Comment by bloons3** [comment]
> If the surgery doesn't have 100% success, it shouldn't have 100% success.

image

**Comment by BrumalLake** [comment]
I didn't actually read the whole pr before this but yeah, it seems that surgeries that inherently had a chance to fail even at a 1x multiplier will still fail

**Comment by BrumalLake** [comment]
Can we get a vote on this or something

**Review by Archanial — APPROVED** [review]

File: code/modules/surgery/surgery_step.dm [hunk 4]
```diff
@@ -55,10 +55,35 @@
 
 	return FALSE
 
+/datum/surgery_step/proc/get_speed_modifier(mob/user, mob/target)
+	var/propability = 0.3
+	var/turf/T = get_turf(target)
+	var/selfpenalty = 0
+	var/sleepbonus = 0
+	if(target == user)
+		if(HAS_TRAIT(user, TRAIT_SELF_AWARE) || locate(/obj/structure/mirror) in view(1, user) || user.get_inactive_held_item() == /obj/item/handmirror)
```
Comment (francinum) [inline code/modules/surgery/surgery_step.dm]: View should only be run if it really has to.
```suggestion
		if(HAS_TRAIT(user, TRAIT_SELF_AWARE) || user.get_inactive_held_item() == /obj/item/handmirror || locate(/obj/structure/mirror) in view(1, user))
```
It's a nitpicky microopt though.

**Review by Archanial — CHANGES_REQUESTED** [review]
Ghetto surgery and self surgery should still have pretty big chance of failure while surgery in relatively good conditions (at least with actual medical tools) could operate on speed modifier alone. 
Also stopping the do after mid surgery should deal small amount of damage with some feedback like "while stopping the surgery, you make a mistake" or something.

**Comment by BrumalLake** [comment]
> 
> 
> Ghetto surgery and self surgery should still have pretty big chance of failure while surgery in relatively good conditions (at least with actual medical tools) could operate on speed modifier alone.

The chances are low enough, and i dont want the chances to be so annoying where you fuck up 9 times in a row in alright conditions with ghetto tools like how it is currently even while the target is dead/sleeping which is a huge .5 bonus

> Also stopping the do after mid surgery should deal small amount of damage with some feedback like "while stopping the surgery, you make a mistake" or something.

wyci, this is about surgery speed, and i dont think people are ignorant enough to not notice that they swapped hands or something during a step, its pointless feedback
people already slash the target if they're holding a scalpel or something and most dangerous steps already have a screw up reaction and doing more damage is annoying

**Comment by Archanial** [comment]
You should fuck up even in good conditions with ghetto tools.

People slash with their tools when they fail surgery. What I am suggesting is they also do it when they break do after.

**Comment by BrumalLake** [comment]
> 
> 
> You should fuck up even in good conditions with ghetto tools.
> 
> People slash with their tools when they fail surgery. What I am suggesting is they also do it when they break do after.

you do mess up
they do slash people when doing that

**Comment by utilbird** [comment]
Not sure what the consensus is on this

**Review by utilbird — APPROVED** [review]

**Pull request merged**